### PR TITLE
 [MM-47303] Migrate /enterprise/extend_session/not_extended_when_disabled e2e tests to TypeScript

### PR DESCRIPTION
**Pull request by hereje**
#### Summary

refactor: migrate not_extended_when_disabled to ts

- migrate js files to ts
- add missing types
- update docs
- fix lint issues

#### Ticket Link

Fixes: #21306

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: e2e-tests/cypress/tests/support/ldap_server_commands.ts [hunk 1]
```diff
@@ -110,3 +134,17 @@ mail: ${user.email}
 userPassword: Password1
 `;
 }
+
+declare global {
```
[the remark this answers is not in the document]
Comment (saturninoabril) [reply e2e-tests/cypress/tests/support/ldap_server_commands.ts:138]: Yes, we're doing it here.

**Review by saturninoabril — APPROVED** [review]
Thanks @hanzei, LGTM except for the one comment about `any`. Approving now given it's to be resolved.

File: e2e-tests/cypress/tests/integration/channels/enterprise/extend_session/not_extended_when_disabled/with_saml_login_spec.ts [hunk 2]
```diff
@@ -24,13 +25,16 @@ describe('Extended Session Length', () => {
...
-    let testTeamId;
-    let testSamlUser;
-    let offTopicUrl;
-    let samlLdapUser;
+    let testTeamId: string;
+    let testSamlUser: UserProfile;
+    let offTopicUrl: string;
+
+    // TODO: improve data type here
+    let samlLdapUser: any;
```
[the remark this answers is not in the document]
Comment (hereje) [reply e2e-tests/cypress/tests/integration/channels/enterprise/extend_session/not_extended_when_disabled/with_saml_login_spec.ts]: I have changed it to type LdapUser

File: e2e-tests/cypress/tests/support/fetch_commands.ts [hunk 3]
```diff
@@ -45,7 +47,8 @@ const mockWebsocketsFn = (win) => {
                 if (this.wrappedSocket) {
                     this.wrappedSocket.send(data);
                 } else {
-                    onerror();
+                    // TODO: define parameters for onerror
```
Comment (M-ZubairAhmed) [inline e2e-tests/cypress/tests/support/fetch_commands.ts:50]: can you remove this todo

**Review by M-ZubairAhmed — CHANGES_REQUESTED** [review]

**Review by M-ZubairAhmed — APPROVED** [review]
This looks good :shipit: Thank you

**Pull request merged**